syntax = "proto3";

package at.tracklab;

message Driver {
  string id = 1;
  string permanentNumber = 2;
  string code = 3;
  string givenName = 4;
  string familyName = 5;
  optional string dateOfBirth = 6;
  optional string countryCode = 7;
  optional string nationality = 8;
  optional string url = 9;
  optional string headshotUrl = 10;

}

message Team {
  string id = 1;
  string name = 2;
  optional string color = 3;
  optional string nationality = 4;
  optional string url = 5;

}

message Circuit {
  string message = 2;

}

message Location {
  string country = 1;
  string locality = 2;
}

message DriverResult {
  Driver driver = 1;
  Team team = 2;

  float position = 3;
  string classifiedPosition = 4;
  float gridPosition = 5;
  float points = 6;
  string status = 7;

  string q1 = 8;
  string q2 = 9;
  string q3 = 10;
  string raceTime = 11;
}

message SessionResult {
  string sessionType = 1;
  string eventName = 2;
  string eventFormat = 3;

  Location location = 4;

  string date = 5;
  int32 year = 6;

  repeated DriverResult driverResults = 7;
}

